crates/ruma-identifiers/src/server_name.rs

```diff
@@ -1,12 +1,32 @@
 //! Matrix-spec compliant server names.
 
+use std::net::Ipv4Addr;
+
 /// A Matrix-spec compliant server name.
 #[repr(transparent)]
 #[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct ServerName(str);
 
 opaque_identifier_validated!(ServerName, ruma_identifiers_validation::server_name::validate);
 
+impl ServerName {
+    /// Returns the server name without the port.
+    pub fn without_port(&self) -> &str {
+        if let Some(end_of_ipv6) = self.0.find(']') {
+            &self.0[0..=end_of_ipv6]
+        } else {
+            // It's not ipv6, so ':' means the port starts
```

Comment:
This is a relatively crude method of determining if it's ipv6 or not, see my comment above, i recommend letting the rust standard library parse this.

Reply:
Rusts parser for Ipv6Addr does not parse the "[" and "]" and also not the port.

Reply:
[SocketAddrV6](https://doc.rust-lang.org/std/net/struct.SocketAddrV6.html)

You can use [SocketAddr::from_str](https://doc.rust-lang.org/std/str/trait.FromStr.html#tymethod.from_str) to parse it, get an error if it failed parsing, gain info if its ipv6, and also get the port and the ip seperately.

Reply:
Ah, the validation logic could use that as well then (I followed that approach):

https://github.com/ruma/ruma/blob/7e3b727d0459b0434d4d0b02ec3fca242c300c44/crates/ruma-identifiers-validation/src/server_name.rs

Reply:
Ah thanks, i'll file an issue for that.